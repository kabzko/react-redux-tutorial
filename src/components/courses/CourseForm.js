import React from "react";
import TextInput from "../common/TextInput";
import SelectInput from "../common/SelectInput";
import PropTypes from "prop-types";

function CourseForm({
    onSave,
    onChange,
    course,
    errors,
    authors,
    saving
}) {
    return (
        <div className="container">
            <h2>{course.id ? "Edit" : "Add"} Course</h2>
            <form onSubmit={onSave}>
                <TextInput
                    id="title"
                    name="title"
                    label="Title"
                    onChange={onChange}
                    value={course.title}
                    error={errors.title}
                />
                <SelectInput 
                    id="authorId"
                    name="authorId"
                    label="Author"
                    onChange={onChange}
                    value={course.authorId || ""}
                    defaultOption="Select Author"
                    options={authors.map(author => ({
                        value: author.id,
                        text: author.name
                    }))}
                    error={errors.author}
                />
                <TextInput
                    id="category"
                    name="category"
                    label="Category"
                    onChange={onChange}
                    value={course.category}
                    error={errors.category}
                />
                <button type="submit" disabled={saving} className="btn btn-primary">
                    {saving ? "Saving..." : "Save"}
                </button>
            </form>
        </div>
    )
}

CourseForm.propTypes = {
    authors: PropTypes.array.isRequired,
    saving: PropTypes.bool,
    course: PropTypes.object.isRequired,
    onSave: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    errors: PropTypes.object
}

CourseForm.defaultProps = {
    errors: {}
}

export default CourseForm;