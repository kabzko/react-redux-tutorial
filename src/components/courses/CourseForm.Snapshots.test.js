import React from "react";
import CourseForm from "./CourseForm";
import renderer from "react-test-renderer";
import { courses, authors } from "../../../tools/mockData";

it("Sets submit button label 'Savings...' when saving is true", () => {
    const tree = renderer.create(
        <CourseForm 
            course={courses[0]}
            authors={authors}
            onChange={jest.fn()}
            onSave={jest.fn()}
            saving
        />
    )

    expect(tree).toMatchSnapshot();
})

it("Sets submit button label 'Save' when saving is false", () => {
    const tree = renderer.create(
        <CourseForm 
            course={courses[0]}
            authors={authors}
            onChange={jest.fn()}
            onSave={jest.fn()}
            saving={false}
        />
    )

    expect(tree).toMatchSnapshot();
})