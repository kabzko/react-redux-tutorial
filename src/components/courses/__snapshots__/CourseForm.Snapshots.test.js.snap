// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets submit button label 'Save' when saving is false 1`] = `
<div
  className="container"
>
  <h2>
    Edit
     Course
  </h2>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="title"
      >
        Title
      </label>
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
      <div
        className="invalid-feedback"
      >
        
      </div>
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="authorId"
      >
        Author
      </label>
      <select
        className="form-select"
        id="authorId"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Author
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
      <div
        className="invalid-feedback"
      >
        
      </div>
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="category"
      >
        Category
      </label>
      <input
        className="form-control"
        id="category"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
      <div
        className="invalid-feedback"
      >
        
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;

exports[`Sets submit button label 'Savings...' when saving is true 1`] = `
<div
  className="container"
>
  <h2>
    Edit
     Course
  </h2>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="title"
      >
        Title
      </label>
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
      <div
        className="invalid-feedback"
      >
        
      </div>
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="authorId"
      >
        Author
      </label>
      <select
        className="form-select"
        id="authorId"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Author
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
      <div
        className="invalid-feedback"
      >
        
      </div>
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="category"
      >
        Category
      </label>
      <input
        className="form-control"
        id="category"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
      <div
        className="invalid-feedback"
      >
        
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Saving...
    </button>
  </form>
</div>
`;
